### ISSUE TITLE
[Accessibility] Form cannot be focused and announced by Narrator automatically when launching app's exe from file explorer

### ISSUE THREAD

**Issue by Olina-Zhang** [dotnet/winforms#8098]
### .NET version

8.0.100-alpha.1.22551.5

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, repro in previous all versions

### Issue description

Form cannot be focused and announced by Narrator automatically when launching app's exe from debug folder

.Net 8.0:
![Form_8 0](https://user-images.githubusercontent.com/26474449/199459985-975bbdb1-7d79-4352-be20-62a328bb71b2.gif)

.Net framework 4.8.1:
![Form_framework481](https://user-images.githubusercontent.com/26474449/199460019-0ae99a21-1f4a-4899-9de0-d45e3151d622.gif)

### Steps to reproduce

1. Create a Winforms .net application with targeting .net 8.0
2. Build application
3. In file explorer, find this application and go to debug->**net8.0-windows** folder
4. Launch Narrator and click app's exe

**Comment by dmitrii-drobotov** [comment]
This is a follow-up on #6896

**Comment by merriemcgaw** [comment]
@NikitaSemenovAkvelon if you can confirm that if there is something focusable on the form that Narrator does pick it up, then I think we can hold off on a deeper investigation for now.

**Comment by NikitaSemenovAkvelon** [comment]
If a form has any focusable element, all works fine.
But the issue is more complex in fact. Narrator (also as NVDA) doesn't announce any events of forms until it is refocused. For a test, I added a custom notification in the OnClick method and it wasn't announced.

**Comment by Tanya-Solyanik** [comment]
>I added a custom notification in the OnClick method and it wasn't announced.

Please clarify your experiment. To invoke OnClick, you have to click on the form and thus put focus on it.... You need notifications on timer to avoid focus. No announcement unless application is focused looks like a "by design" feature for certain types of notifications. For example, when I'm working in VisualStudio I might want or might not want to receive notifications from teams if teams does not have focus.

>Narrator (also as NVDA) doesn't announce any events of forms until it is refocused

If you are using UIARaiseNotificationEvent, then announcement depends on how narrator is configures to interpret this value -  https://learn.microsoft.com/en-us/windows/win32/api/uiautomationcore/ne-uiautomationcore-notificationprocessing

**Comment by merriemcgaw** [comment]
Let's won't fix this issue for now, because there are essentially no customers impacting scenarios where this will be a problem. We will reactivate if needed.

**Comment by NikitaSemenovAkvelon** [comment]
The result of an investigation that might be useful in the future:

1. The bug is reproduced via a standalone application or via VS (no matter with or without debugging).
2. The bug is reproduced only for the main form (if you open another empty form in an application, it works fine).
3. The bug is reproduced only with the UIA provider (`SupportsUIAProvider => true`). MSAA works as expected.
4. The bug isn't reproduced with WPF.
5. No notifications aren't announced by Narrator or NVDA until a screen reader got focused on the main form. The most simple way to set focus is to remove focus from the form and set it again. Gif below is showing how it works:
![VACpxvAjEo](https://user-images.githubusercontent.com/109065597/202113779-58fe0768-3848-4f4b-b412-8ffd3781d2b5.gif)
For testing I'd added override of OnClick method:

> protected override void OnClick(EventArgs e)
>         {
>             base.OnClick(e);
>             Text = DateTime.Now.ToLongTimeString() + IsAccessibilityObjectCreated.ToString();
>             AccessibilityObject.RaiseAutomationPropertyChangedEvent(UiaCore.UIA.NamePropertyId, Text, Text);
>             AccessibilityObject.RaiseAutomationNotification(Automation.AutomationNotificationKind.Other,
>                     Automation.AutomationNotificationProcessing.All,
>                     "Test");
>         }

You can see in the gif, that `Name` of the form was changed and `IsAccessibilityObjectCreated` returned `true`, but Narrator didn't announced it until it got focused.

6. A notification `AutomationFocusChangedEventId` is missed in AccEvent. But the result of method is `true`.
https://github.com/dotnet/winforms/blob/be03f45ccd99fded5a317daf825dc8bb6bbabb5c/src/System.Windows.Forms/src/System/Windows/Forms/Form.cs#L3108
UIA provider:
![MicrosoftTeams-image](https://user-images.githubusercontent.com/109065597/202115708-1eb56daa-9738-436b-bb81-6ded275947dc.png)
MSAA:
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/109065597/202115773-90c8c9d3-d76c-4102-bf6c-937934e9e558.png)
7. It has an unstable workaround. If the call is mentioned in the previous point call in another thread after some delay, Narrator will get the focus and announce the event. But it is unstable and works on one of the four-five tries. Code that I used to test:

> Task.Run(async () =>
>                 {
>                     await Task.Delay(1000).ConfigureAwait(false);
>                     AccessibilityObject.RaiseAutomationEvent(UiaCore.UIA.AutomationFocusChangedEventId);
>                 });

*_Time to delay isn't relevant to the result. I tried 500, 1000, and 2000. The result wasn't changed._
8. Adding a call `Application.DoEvents()` before creating of AO wasn't help too.
